src/NuGet.Core/NuGet.Commands/PackCollectorLogger.cs

```diff
@@ -14,14 +14,22 @@ public class PackCollectorLogger : LoggerBase
         private readonly ConcurrentQueue<ILogMessage> _errors;
         private ILogger _innerLogger;
 
+        // Project level warnings properties
         public WarningProperties WarningProperties { get; set; }
 
+        // Package specific warning property
+        internal PackCommand.PackageSpecificWarningProperties PackageSpecificWarningProperties { get; set; }
```

Comment:
I might've made this recommendation, but this might be private. 

It isn't used outside of this class.